### DISCUSSION TITLE
New Field: Vector

### DISCUSSION THREAD

**Discussion by Livog** [payloadcms/payload#8403] [Feature Requests & Ideas]
With the rapid growth of AI, there is increased interest in storing vectorized data for AI-driven searches and other machine learning applications. Both MongoDB and PostgreSQL (via the pg_vector extension) offer support for vector columns.
MongoDB Docs: https://www.mongodb.com/docs/atlas/atlas-vector-search/vector-search-type/

Implementing native vector field in Payload CMS would provide better performance compared to using the text field, especially as the users databases grows.

I'm very curious if this is possible and if so, is'it a large, medium or small task?

**Comment by cgilly2fast** [comment]
Its possible its an enterprise feature for payload: https://payloadcms.com/enterprise/ai-framework

**Comment by donbellamy** [comment]
Looking for this functionality as well.

**Reply by cgilly2fast** [reply]
My work around is the json field, where I store the array of numbers for the vector. Then I set up the vector index in mongo atlas db
